specification/azurestack/resource-manager/Microsoft.AzureStack/stable/2017-06-01/AzureStack.json

```diff
@@ -44,6 +44,53 @@
           "nextLinkName": "nextLink"
         }
       }
+    },
+    "/providers/Microsoft.AzureStack/cloudmanifestfile": {
```

Comment:
>cloudmanifestfile [](start = 37, length = 17)

As per the RPC, resource type names should be plural. If it's a singleton, the correct pattern would be to implement the following routes:
- `/providers/Microsoft.AzureStack/cloudmanifestfiles`
- `/providers/Microsoft.AzureStack/cloudmanifestfiles/default`

`default` can be substituted with `master`, `current`, or `latest` depending on the requirements.

Reply:
Fixed

Reply:
You need the list route also (`/providers/Microsoft.AzureStack/cloudmanifestfiles`)

---
In reply to: [379690800](https://github.com/Azure/azure-rest-api-specs/pull/8240#discussion_r379690800) [](ancestors = 379690800)

Reply:
Do I have to implement the API for list? We don't have a use case for that.

Reply:
Yeah :(. GET verb in ARM RPC is reserved for resources, which for a read-only resource like this would require LIST and GET at minimum. The other alternative would be to make it a POST action.

Reply:
For some reason, I couldn't get proxy action work on tenant level. And this should not be a POST. I'll probably implement the list same as GET default.